### PR TITLE
Add new test for JDK16/17, Update test cases for JDK18(JEP-400) and Update codepoint/unicode for JDK19

### LINKED ISSUE #3350: Add new test for JDK16/17, Update test cases for JDK18(JEP-400) and Update codepoint/unicode for JDK19
1. I created tests for new features in JDK16/17.

- pattern_matching_instanceof
- sealed_classes
- record

2. I updated almost every test for JDK18(JEP-400).
3. I updated Unicode data in codepoint/unicode for JDK19.

### PR DESCRIPTION
**Pull request by KKES335**
Fixes #3350 

Signed-off-by: Kentaro Kuramochi (kkes335@jp.ibm.com)

I created tests for new features in JDK16/17.
-  pattern_matching_instanceof
-  sealed_classes
-  record

I updated almost every test for JDK18(JEP-400).
I updated Unicode data in codepoint/unicode for JDK19.

### CODE REVIEW

**Comment by KKES335** [comment]
Here are Grinder_Simple results

x86-64_windows, JDK_VERSION 19, JDK_IMPL hotspot
https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder_Simple/298/tapResults/
452 tests, 88 ok, 0 not ok, 364 skipped, 0 ToDo, 0 Bail Out!

ppc64_aix, JDK_VERSION 19, JDK_IMPL hotspot
https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder_Simple/299/tapResults/
452 tests, 272 ok, 0 not ok, 180 skipped, 0 ToDo, 0 Bail Out!

x86-64_linux, JDK_VERSION 19, JDK_IMPL hotspot
https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder_Simple/300/tapResults/
452 tests, 104 ok, 0 not ok, 348 skipped, 0 ToDo, 0 Bail Out!

**Review by smlambert — APPROVED** [review]
thank you @KKES335 !

File: functional/MBCS_Tests/pattern_matching_instanceof/build.xml [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+	<!--Properties for this particular build-->
+	<property name="src" location="./src" />
+	<property name="build" location="./bin" />
+	
+	<target name="init">
+		<mkdir dir="${DEST}" />
+		<mkdir dir="${build}" />
+	</target>
+
+	<target name="compile" depends="init" description="Using java${JDK_VERSION} to compile the source">
```
Comment (sophia-guo) [inline functional/MBCS_Tests/pattern_matching_instanceof/build.xml:36]: This is for jdk17+? If that is the case I would suggest do a if condition in build task. That is 

```
	<target name="build">
		<if>
			<not>
				<matches string="${JDK_VERSION}" pattern="^(8|9|10|11|12|13|14|15|16)$$" />
			</not>
			<then>
				<antcall target="clean" inheritall="true" />
			</then>
		</if>
	</target>
```

File: functional/MBCS_Tests/pattern_matching_instanceof/build.xml [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+		<echo>Ant version is ${ant.version}</echo>
+		<echo>============COMPILER SETTINGS============</echo>
+		<echo>===fork:				yes</echo>
+		<echo>===executable:			${compiler.javac}</echo>
+		<echo>===debug:				on</echo>
+		<echo>===destdir:				${DEST}</echo>
+		<if>
```
Comment (sophia-guo) [inline functional/MBCS_Tests/pattern_matching_instanceof/build.xml:43]: See my former comment so we don't need to check the version at compile task.

**Comment by KKES335** [comment]
@sophia-guo 
Thank you for your comment.
I fixed some build.xmls as you mentioned and I confirmed they worked fine.

**Review by sophia-guo — APPROVED** [review]
LGTM. Thanks @KKES335!
